### PR TITLE
Improve chat

### PR DESCRIPTION
**Pull request by joseivanlopez**
Some chat improvements:

* Layout is remembered (open/closed chat).
* Show number of unread chat messages.

This PR fixes part of https://github.com/jangouts/jangouts/issues/331.

### CODE REVIEW

File: src/components/Chat/Toggler/Toggler.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+function Toggler() {
+  const dispatch = useDispatch();
+  const { settings } = useSelector((state) => state.room);
+
+  let title;
+  let classes;
+
+  if (settings.chatOpen) {
```
Comment (mvidner) [inline src/components/Chat/Toggler/Toggler.js:28]: I don't understand how `settings` can have a value now.
It comes from `useSelector`, which is a library function, not having access to any of our data, and we pass it only a trivial lambda.
Redux magic?

Comment (mvidner) [reply src/components/Chat/Toggler/Toggler.js:28]: Well maybe it's why the tests fail too?

**Comment by imobachgs** [comment]
Actually, that's a good question. And yes, there is some Redux magic there. I will try to explain what I know. @joseivanlopez might be interested in this topic too, as we were talking about that recently.

First of all, I would like to recommend installing the [Redux DevTools](https://github.com/reduxjs/redux-devtools). IMHO, it is a must-have when working with Redux, as it allows you to see Redux actions and how they change the state.

Now, let's start from the beginning. As you know, Redux keeps our application state in a single object. Initially, it looks like this:

```javascript
{
  room: {
    settings: {},
    loggedIn: false,
    loggingIn: false
  },
  participants: {},
  messages: [],
  config: {},
  notifications: {
    notifications: [],
    blocklist: {}
  }
}
```

You can read the [Initializing State](https://redux.js.org/usage/structuring-reducers/initializing-state) to know more about how the store is initialized. In our case, we pass the initial state of each section to its corresponding reducer. See [src/state/ducks/room.js](https://github.com/jangouts/jangouts/blob/1e5ede00368f321db9128470d99ae39463431ab1/src/state/ducks/room.js#L82) for an example. When the application starts, Redux fires an `@@INIT` action which causes all reducers to run and initialize the state.

Later, the state might look like this:

```javascript
{
  room: {
    settings: {
      _username: 'imo-laptop',
      _roomId: '2003'
    },
    loggedIn: false,
    loggingIn: false
  },
  participants: {},
  messages: [],
  config: {
    janusWithCredentials: true,
    joinUnmutedLimit: 3,
    videoThumbnails: true,
    janusServer: 'wss://li2023-182.members.linode.com/janus/'
  },
  notifications: {
    notifications: [],
    blocklist: {}
  }
}
```

According to [useSelector documentation](https://react-redux.js.org/api/hooks#useselector), it *allows you to extract data from the Redux store state, using a selector function*. So it receives the whole state and it just returns the part you are interested in.

Finally, you might wonder why `_username` and `_roomId` are prepended with an underscore. The reason is that they are private attributes of the `UserSettings` class. And that takes me to the next point: *it is highly recommended that you only put plain serializable objects, arrays, and primitives into your store* ([source](https://redux.js.org/faq/organizing-state#can-i-put-functions-promises-or-other-non-serializable-items-in-my-store-state)). I am sorry, @joseivanlopez, but I forgot about that detail 😞. But we can adapt #374 easily 😉.

File: src/components/Chat/Toggler/Toggler.js (hunk 1 above)
Comment (imobachgs) [reply src/components/Chat/Toggler/Toggler.js:28]: Actually, that's a good question. And yes, there is some Redux magic there. I will try to explain what I know. @joseivanlopez might be interested in this topic too, as we were talking about that recently.

First of all, I would like to recommend installing the [Redux DevTools](https://github.com/reduxjs/redux-devtools). IMHO, it is a must-have when working with Redux, as it allows you to see Redux actions and how they change the state.

Now, let's start from the beginning. As you know, Redux keeps our application state in a single object. Initially, it looks like this:

```javascript
{
  room: {
    settings: {},
    loggedIn: false,
    loggingIn: false
  },
  participants: {},
  messages: [],
  config: {},
  notifications: {
    notifications: [],
    blocklist: {}
  }
}
```

You can read the [Initializing State](https://redux.js.org/usage/structuring-reducers/initializing-state) to know more about how the store is initialized. In our case, we pass the initial state of each section to its corresponding reducer. See [src/state/ducks/room.js](https://github.com/jangouts/jangouts/blob/1e5ede00368f321db9128470d99ae39463431ab1/src/state/ducks/room.js#L82) for an example. When the application starts, Redux fires an `@@INIT` action which causes all reducers to run and initialize the state.

Later, the state might look like this:

```javascript
{
  room: {
    settings: {
      _username: 'imo-laptop',
      _roomId: '2003'
    },
    loggedIn: false,
    loggingIn: false
  },
  participants: {},
  messages: [],
  config: {
    janusWithCredentials: true,
    joinUnmutedLimit: 3,
    videoThumbnails: true,
    janusServer: 'wss://li2023-182.members.linode.com/janus/'
  },
  notifications: {
    notifications: [],
    blocklist: {}
  }
}
```

According to [useSelector documentation](https://react-redux.js.org/api/hooks#useselector), it *allows you to extract data from the Redux store state, using a selector function*. So it receives the whole state and it just returns the part you are interested in.

Finally, you might wonder why `_username` and `_roomId` are prepended with an underscore. The reason is that they are private attributes of the `UserSettings` class. And that takes me to the next point: *it is highly recommended that you only put plain serializable objects, arrays, and primitives into your store* ([source](https://redux.js.org/faq/organizing-state#can-i-put-functions-promises-or-other-non-serializable-items-in-my-store-state)). I am sorry, @joseivanlopez, but I forgot about that detail 😞. But we can adapt #374 easily 😉.

Comment (joseivanlopez) [reply src/components/Chat/Toggler/Toggler.js:28]: Fixed.

**Review by dgdavid — APPROVED** [review]
LGTM

I left two neat pickings, but you can perfectly ignore them :stuck_out_tongue_closed_eyes:

**Pull request merged**